### PR TITLE
[MRG] f-string updates for issue #29547

### PR DESCRIPTION
**Pull request by abbiepopa**
Addresses, in part, https://github.com/pandas-dev/pandas/issues/29547 

…ape/pivot, reshape/reshape

- [ x] xref #29547 
- [n/a ] tests added / passed
- [x ] passes `black pandas`
- [ x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [n/a ] whatsnew entry

### CODE REVIEW

File: pandas/core/reshape/concat.py [hunk 1]
```diff
@@ -352,8 +352,8 @@ def __init__(
         for obj in objs:
             if not isinstance(obj, (Series, DataFrame)):
                 msg = (
-                    "cannot concatenate object of type '{typ}'; "
-                    "only Series and DataFrame objs are valid".format(typ=type(obj))
+                    f"cannot concatenate object of type '{type(obj)}'; "
+                    f"only Series and DataFrame objs are valid"
```
Comment (WillAyd) [inline pandas/core/reshape/concat.py]:
```suggestion
                    "only Series and DataFrame objs are valid"
```

Don't need the f-prefix for lines with no replacement

File: pandas/core/reshape/melt.py [hunk 2]
```diff
@@ -417,9 +417,7 @@ def wide_to_long(
     """
 
     def get_var_names(df, stub: str, sep: str, suffix: str) -> List[str]:
-        regex = r"^{stub}{sep}{suffix}$".format(
-            stub=re.escape(stub), sep=re.escape(sep), suffix=suffix
-        )
+        regex = r"^{re.escape(stub)}{re.escape(sep)}{suffix}$"
```
Comment (WillAyd) [inline pandas/core/reshape/melt.py]:
```suggestion
        regex = fr"^{re.escape(stub)}{re.escape(sep)}{suffix}$"
```

Missing the prefix here

File: pandas/core/reshape/merge.py [hunk 3]
```diff
@@ -610,9 +610,10 @@ def __init__(
         # warn user when merging between different levels
         if _left.columns.nlevels != _right.columns.nlevels:
             msg = (
-                "merging between different levels can give an unintended "
-                "result ({left} levels on the left, {right} on the right)"
-            ).format(left=_left.columns.nlevels, right=_right.columns.nlevels)
+                f"merging between different levels can give an unintended "
```
Comment (WillAyd) [inline pandas/core/reshape/merge.py]:
```suggestion
                "merging between different levels can give an unintended "
```

File: pandas/core/reshape/merge.py [hunk 4]
```diff
@@ -678,8 +679,8 @@ def _indicator_pre_merge(
         for i in ["_left_indicator", "_right_indicator"]:
             if i in columns:
                 raise ValueError(
-                    "Cannot use `indicator=True` option when "
-                    "data contains a column named {name}".format(name=i)
+                    f"Cannot use `indicator=True` option when "
```
Comment (WillAyd) [inline pandas/core/reshape/merge.py]:
```suggestion
                    "Cannot use `indicator=True` option when "
```

File: pandas/core/reshape/merge.py [hunk 5]
```diff
@@ -1184,14 +1185,11 @@ def _validate_specification(self):
...
                     raise MergeError(
-                        "No common columns to perform merge on. "
-                        "Merge options: left_on={lon}, right_on={ron}, "
-                        "left_index={lidx}, right_index={ridx}".format(
-                            lon=self.left_on,
-                            ron=self.right_on,
-                            lidx=self.left_index,
-                            ridx=self.right_index,
-                        )
+                        f"No common columns to perform merge on. "
```
Comment (WillAyd) [inline pandas/core/reshape/merge.py]:
```suggestion
                        "No common columns to perform merge on. "
```

File: pandas/core/reshape/merge.py [hunk 6]
```diff
@@ -1680,8 +1672,9 @@ def _get_merge_keys(self):
 
         # validate allow_exact_matches
         if not is_bool(self.allow_exact_matches):
-            msg = "allow_exact_matches must be boolean, passed {passed}"
-            raise MergeError(msg.format(passed=self.allow_exact_matches))
+            msg = f"allow_exact_matches must be boolean, " \
```
Comment (WillAyd) [inline pandas/core/reshape/merge.py]: We don't use explicit line continuation, so just use parentheses as needed

File: pandas/core/reshape/reshape.py [hunk 7]
```diff
@@ -873,15 +873,13 @@ def get_dummies(
...
-                "length of the columns being encoded ({len_enc})."
-            )
 
             if is_list_like(item):
                 if not len(item) == data_to_encode.shape[1]:
-                    len_msg = len_msg.format(
-                        name=name, len_item=len(item), len_enc=data_to_encode.shape[1]
+                    len_msg = (
+                        f"Length of '{name}' ({len(item)}) did not match the "
+                        f"length of the columns being encoded "
```
Comment (WillAyd) [inline pandas/core/reshape/reshape.py]:
```suggestion
                        "length of the columns being encoded "
```

**Review by WillAyd — CHANGES_REQUESTED** [review]
Thanks for the PR! Mostly minor edits

**Comment by abbiepopa** [comment]
Thank you @WillAyd , I covered the changes you requested. I'm having some trouble figuring out why I'm failing the current tests. LMK if it's obvious to you, otherwise I will continue investigating.

**Comment by MarcoGorelli** [comment]
Thanks, Abbie!

> I'm having some trouble figuring out why I'm failing the current tests

If you click on 'details' next to, say, "CI / Checks", you'll see an 'X' near 'linting'. If you expand it, you'll see
```
black --version
black, version 19.10b0
Checking black formatting
would reformat /home/runner/work/pandas/pandas/pandas/core/reshape/concat.py
would reformat /home/runner/work/pandas/pandas/pandas/core/reshape/melt.py
would reformat /home/runner/work/pandas/pandas/pandas/core/reshape/pivot.py
would reformat /home/runner/work/pandas/pandas/pandas/core/reshape/merge.py
Oh no! 💥 💔 💥
4 files would be reformatted, 923 files would be left unchanged.
```
So for a start, you need to run `black` again any file you've changed. For example,

```
black pandas/core/reshape/concat.py
```

The [contributing guide](https://pandas.pydata.org/pandas-docs/stable/development/contributing.html#contributing-to-the-code-base) has some more info about this.

**Comment by abbiepopa** [comment]
Thank you, @MarcoGorelli ! Looks like the latest failure is a timeout. Is there a way to retrigger the test that timed out?

**Comment by abbiepopa** [comment]
Yay! Thanks @WillAyd and @MarcoGorelli . LMK if you need additional changes etc.

File: pandas/core/reshape/merge.py [hunk 8]
```diff
@@ -1708,20 +1701,23 @@ def flip(xs):
         tolerance = self.tolerance
 
         # we require sortedness and non-null values in the join keys
-        msg_sorted = "{side} keys must be sorted"
-        msg_missings = "Merge keys contain null values on {side} side"
+        def _msg_sorted(side):
```
Comment (jreback) [inline pandas/core/reshape/merge.py]: i would revert this section, it is more clear the existing way

**Review by jreback — COMMENTED** [review]
comment, pls ping on green.

File: pandas/core/reshape/concat.py [hunk 9]
```diff
@@ -622,11 +621,7 @@ def _make_concat_multiindex(indexes, keys, levels=None, names=None) -> MultiInde
                 try:
                     i = level.get_loc(key)
                 except KeyError:
-                    raise ValueError(
-                        "Key {key!s} not in level {level!s}".format(
-                            key=key, level=level
-                        )
-                    )
+                    raise ValueError(f"Key {key!s} not in level {level!s}")
```
Comment (MomIsBestFriend) [inline pandas/core/reshape/concat.py]: I *think* you can get rid of the ```!s``` just have it as

```python
raise ValueError(f"Key {key} not in level {level}")
```

File: pandas/core/reshape/merge.py [hunk 10]
```diff
@@ -1680,8 +1670,11 @@ def _get_merge_keys(self):
 
         # validate allow_exact_matches
         if not is_bool(self.allow_exact_matches):
-            msg = "allow_exact_matches must be boolean, passed {passed}"
-            raise MergeError(msg.format(passed=self.allow_exact_matches))
+            msg = (
+                f"allow_exact_matches must be boolean, "
```
Comment (MomIsBestFriend) [inline pandas/core/reshape/merge.py]: Can you please remove the ```f``` here?

File: pandas/tests/reshape/merge/test_merge.py [hunk 11]
```diff
@@ -370,11 +370,9 @@ def test_no_overlap_more_informative_error(self):
         df2 = DataFrame({"y": ["b", "c"]}, index=[dt, dt])
 
         msg = (
-            "No common columns to perform merge on. "
-            "Merge options: left_on={lon}, right_on={ron}, "
-            "left_index={lidx}, right_index={ridx}".format(
-                lon=None, ron=None, lidx=False, ridx=False
-            )
+            f"No common columns to perform merge on. "
```
Comment (MomIsBestFriend) [inline pandas/tests/reshape/merge/test_merge.py]: Can you remove the ```f```` here?

**Comment by abbiepopa** [comment]
Thanks @jreback and @MomIsBestFriend ! I updated based on your reviews.

**Review by WillAyd — APPROVED** [review]

**Comment by WillAyd** [comment]
Great thanks @abbiepopa

**Pull request merged**